File: src/Commands/QuitCommand.ts
import { Commands } from './Commands';
import { UnicastMpv } from '../UnicastMpv';
import { tuple } from '../Schema';

export class QuitCommand extends Commands {
    constructor ( server : UnicastMpv ) {
        super( server );

        this.register( 'quit', tuple( [] ), this.quit.bind( this ) );
    }

    public async quit () {
        if ( this.server.player.mpv.isRunning() ) {
            await this.server.player.mpv.quit();
        }
    }
}

File: pkg/commands/quit_command.go
package commands

import (
	"github.com/unicast/unicast-mpv/pkg/schema"
)

type QuitCommand struct {
	*CommandRegistry
}

func NewQuitCommand(registry *CommandRegistry) *QuitCommand {
	qc := &QuitCommand{CommandRegistry: registry}

	qc.Register("quit", schema.Tuple(), qc.quit)

	return qc
}

func (qc *QuitCommand) quit(args []interface{}) (interface{}, error) {
	if qc.mpv.IsRunning() {
		if err := qc.mpv.Quit(); err != nil {
			return nil, err
		}
	}
	return nil, nil
}